compiler/rustc_lint/src/context.rs

```diff
@@ -804,8 +804,6 @@ pub trait LintContext {
                                 db.span_suggestion(value_span, "there is a expected value with a similar name", format!("\"{best_match}\""), Applicability::MaybeIncorrect);
 
                             }
-                        } else if name == sym::feature && is_from_cargo {
```

Comment:
unrelated comment, but I feel like all of this logic should be moved to a separate module instead of being in this big match

Reply:
I've put up https://github.com/rust-lang/rust/pull/119443 to address this suggestion